synapse/config/registration.py

```diff
@@ -196,7 +196,9 @@ def read_config(self, config, **kwargs):
 
         self.inhibit_user_in_use_error = config.get("inhibit_user_in_use_error", False)
 
-    def generate_config_section(self, generate_secrets=False, **kwargs):
+    def generate_config_section(
+        self, generate_secrets: Optional[bool] = False, **kwargs: Any
```

Comment:
I think this should be a `bool`? I don't see any place in the code passing `None` for `generate_secrets`.

Reply:
I think I just based this on what `generate_config` has for type hints... let me double check.